packages/treasury/test/test-interest.js

```diff
@@ -28,249 +33,283 @@ test('too soon', async t => {
-    10n,
-    5n,
+    annualRate,
+    ONE_MONTH,
+    ONE_DAY,
   );
   let debtStatus = {
     newDebt: amountMath.make(100000n, brand),
-    latestInterestUpdate: 10n,
+    interest: amountMath.make(0n, brand),
+    latestInterestUpdate: ONE_DAY,
   };
-  const after10 = {
-    latestInterestUpdate: 10n,
+  const afterOneDay = {
+    latestInterestUpdate: ONE_DAY,
     interest: amountMath.make(0n, brand),
     newDebt: amountMath.make(100000n, brand),
   };
-  t.deepEqual(calculator.calculate(debtStatus, 11n), after10);
-  t.deepEqual(calculator.calculate(debtStatus, 13n), after10);
-  t.deepEqual(calculator.calculate(debtStatus, 15n), after10);
-  t.deepEqual(calculator.calculate(debtStatus, 17n), after10);
-  t.deepEqual(calculator.calculate(debtStatus, 19n), after10);
+  t.deepEqual(calculator.calculate(debtStatus, 4n * ONE_DAY), afterOneDay);
+  t.deepEqual(calculator.calculate(debtStatus, 13n * ONE_DAY), afterOneDay);
+  t.deepEqual(calculator.calculate(debtStatus, 15n * ONE_DAY), afterOneDay);
+  t.deepEqual(calculator.calculate(debtStatus, 25n * ONE_DAY), afterOneDay);
+  t.deepEqual(calculator.calculate(debtStatus, 29n * ONE_DAY), afterOneDay);
 
-  const after23 = {
-    latestInterestUpdate: 20n,
-    interest: amountMath.make(1000n, brand),
-    newDebt: amountMath.make(101000n, brand),
+  const afterThreeDays = {
+    latestInterestUpdate: ONE_MONTH + ONE_DAY,
+    interest: amountMath.make(203n, brand),
+    newDebt: amountMath.make(100203n, brand),
   };
-  t.deepEqual(calculator.calculate(debtStatus, 23n), after23);
+  t.deepEqual(
+    calculator.calculate(debtStatus, ONE_DAY + ONE_MONTH),
+    afterThreeDays,
+  );
+
   debtStatus = {
     newDebt: amountMath.make(101000n, brand),
     latestInterestUpdate: 20n,
+    interest: amountMath.make(0n, brand),
   };
-
-  const after25 = {
+  const afterOneMonth = {
     latestInterestUpdate: 20n,
     interest: amountMath.make(0n, brand),
     newDebt: amountMath.make(101000n, brand),
   };
-  t.deepEqual(calculator.calculate(debtStatus, 27n), after25);
-  t.deepEqual(calculator.calculate(debtStatus, 29n), after25);
-  t.deepEqual(calculator.calculate(debtStatus, 30n), {
-    latestInterestUpdate: 30n,
-    interest: amountMath.make(1010n, brand),
-    newDebt: amountMath.make(102010n, brand),
+  t.deepEqual(calculator.calculate(debtStatus, ONE_MONTH), afterOneMonth);
+  t.deepEqual(calculator.calculate(debtStatus, ONE_MONTH + 10n), afterOneMonth);
+  t.deepEqual(calculator.calculate(debtStatus, ONE_MONTH + 20n), {
+    latestInterestUpdate: 20n + ONE_MONTH,
+    interest: amountMath.make(205n, brand),
+    newDebt: amountMath.make(101205n, brand),
   });
 });
 
 // 2.5 % APR charged weekly, charged daily
 test('basic charge reasonable numbers', async t => {
   const { brand } = makeIssuerKit('ducats');
   const annualRate = makeRatio(250, brand, 10000n);
-  const ONE_DAY = 60n * 60n * 24n;
-  const ONE_MONTH = ONE_DAY * 30n;
   const START_TIME = 1617734746n;
```

Comment:
Comment: This is a UNIX epoch time. 
Please say what the time is supposed to be.

Reply:
Tuesday April 6th 2021 at 11:45am PT